programmers/난이도별/level01.모의고사/rockmiin.py

```diff
@@ -0,0 +1,28 @@
+def solution(answers):
+    cnt=[0, 0, 0]
+    ret_value=[]
+    m1=[1, 2, 3, 4, 5]
+    m2=[2, 1, 2, 3, 2, 4, 2, 5]
+    m3=[3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
+    for i, j in enumerate(answers):
+        if answers[i] == m1[i%5]:
+            cnt[0]+=1
+        if answers[i] == m2[i%8]:
+            cnt[1]+=1
+        if answers[i] == m3[i%10]:
+            cnt[2]+=1
```

Comment:
```suggestion
    m=[[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    for i in range(len(answers)):
        for j, v in enumerate(m):
            if answers[i] == v[i % len(v)]:
                cnt[j]+=1
```
2차원 배열을 사용하면 이렇게 응용할 수 있습니다!

혹은 딕셔너리를 사용한다면 다음과 같이 풀이할 수 있습니다.
```
    ret_value=[]
    m=[
        { "pattern": [1, 2, 3, 4, 5], "size": 5, "cnt": 0 },
        { "pattern": [2, 1, 2, 3, 2, 4, 2, 5], "size": 8, "cnt": 0 },
        { "pattern": [3, 3, 1, 1, 2, 2, 4, 4, 5, 5], "size": 10, "cnt": 0 },
    ]
    for i in range(len(answers)):
        for v in m:
            if answers[i] == v['pattern'][i % v['size']]:
                v['cnt'] += 1

    cnt = [ v['cnt'] for v in m ];
    maxValue = max(cnt)
    for k, v in enumerate(cnt):
        if maxValue == v:
            ret_value.append(k + 1)
    return ret_value
```

Reply:
감사합니다 하나 또 배워갑니다 !! :)